File: worktree-pipeline/src/main/java/com/smartwallet/repository/TransactionRepository.java
package com.smartwallet.repository;

import com.smartwallet.entity.Transaction;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface TransactionRepository extends JpaRepository<Transaction, Long> {
    @Query("SELECT t FROM PortfolioTransaction t WHERE t.asset.wallet.user.id = :userId ORDER BY t.createdAt DESC")
    List<Transaction> findByUserId(Long userId);
    @Query("SELECT t FROM PortfolioTransaction t WHERE t.asset.id = :assetId ORDER BY t.createdAt DESC")
    List<Transaction> findByAssetIdOrderedByDateDesc(Long assetId);
}

File: src/main/kotlin/com/smartwallet/portfolio/repository/TransactionRepository.kt
package com.smartwallet.portfolio.repository

import com.smartwallet.portfolio.model.Transaction
import com.smartwallet.portfolio.model.PortfolioAsset
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface TransactionRepository : JpaRepository<Transaction, Long> {
    fun findAllByAsset(asset: PortfolioAsset): List<Transaction>
}
